File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/ServerIpmiLiveInfo.cpp


#include "ServerIpmiLiveInfo.h"

using namespace Tiny;

ServerIpmiLiveInfo::ServerIpmiLiveInfo()
{
	text = std::string();
	public_ip = std::string();
	allowed_ip = std::string();
	client_username = std::string();
	client_password = std::string();
}

ServerIpmiLiveInfo::ServerIpmiLiveInfo(std::string jsonString)
{
	this->fromJson(jsonString);
}

ServerIpmiLiveInfo::~ServerIpmiLiveInfo()
{

}

void
ServerIpmiLiveInfo::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *textKey = "text";

    if(object.has_key(textKey))
    {
        bourne::json value = object[textKey];



        jsonToValue(&text, value, "std::string");


    }

    const char *public_ipKey = "public_ip";

    if(object.has_key(public_ipKey))
    {
        bourne::json value = object[public_ipKey];



        jsonToValue(&public_ip, value, "std::string");


    }

    const char *allowed_ipKey = "allowed_ip";

    if(object.has_key(allowed_ipKey))
    {
        bourne::json value = object[allowed_ipKey];



        jsonToValue(&allowed_ip, value, "std::string");


    }

    const char *client_usernameKey = "client_username";

    if(object.has_key(client_usernameKey))
    {
        bourne::json value = object[client_usernameKey];



        jsonToValue(&client_username, value, "std::string");


    }

    const char *client_passwordKey = "client_password";

    if(object.has_key(client_passwordKey))
    {
        bourne::json value = object[client_passwordKey];



        jsonToValue(&client_password, value, "std::string");


    }


}

bourne::json
ServerIpmiLiveInfo::toJson()
{
    bourne::json object = bourne::json::object();





    object["text"] = getText();






    object["public_ip"] = getPublicIp();






    object["allowed_ip"] = getAllowedIp();






    object["client_username"] = getClientUsername();






    object["client_password"] = getClientPassword();



    return object;

}

std::string
ServerIpmiLiveInfo::getText()
{
	return text;
}

void
ServerIpmiLiveInfo::setText(std::string text)
{
	this->text = text;
}

std::string
ServerIpmiLiveInfo::getPublicIp()
{
	return public_ip;
}

void
ServerIpmiLiveInfo::setPublicIp(std::string public_ip)
{
	this->public_ip = public_ip;
}

std::string
ServerIpmiLiveInfo::getAllowedIp()
{
	return allowed_ip;
}

void
ServerIpmiLiveInfo::setAllowedIp(std::string allowed_ip)
{
	this->allowed_ip = allowed_ip;
}

std::string
ServerIpmiLiveInfo::getClientUsername()
{
	return client_username;
}

void
ServerIpmiLiveInfo::setClientUsername(std::string client_username)
{
	this->client_username = client_username;
}

std::string
ServerIpmiLiveInfo::getClientPassword()
{
	return client_password;
}

void
ServerIpmiLiveInfo::setClientPassword(std::string client_password)
{
	this->client_password = client_password;
}

File: OPENAPI-REST-API/swagger-client/spring/src/main/java/io/swagger/model/ServerIpmiLiveInfo.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import org.springframework.validation.annotation.Validated;
import org.openapitools.jackson.nullable.JsonNullable;
import io.swagger.configuration.NotUndefined;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Information about the IPMI connection.
 */
@Schema(description = "Information about the IPMI connection.")
@Validated
@NotUndefined



public class ServerIpmiLiveInfo   {
  @JsonProperty("text")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String text = null;

  @JsonProperty("public_ip")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String publicIp = null;

  @JsonProperty("allowed_ip")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String allowedIp = null;

  @JsonProperty("client_username")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String clientUsername = null;

  @JsonProperty("client_password")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String clientPassword = null;


  public ServerIpmiLiveInfo text(String text) { 

    this.text = text;
    return this;
  }

  /**
   * Get text
   * @return text
   **/
  
  @Schema(description = "")
  
  public String getText() {  
    return text;
  }



  public void setText(String text) { 
    this.text = text;
  }

  public ServerIpmiLiveInfo publicIp(String publicIp) { 

    this.publicIp = publicIp;
    return this;
  }

  /**
   * Get publicIp
   * @return publicIp
   **/
  
  @Schema(description = "")
  
  public String getPublicIp() {  
    return publicIp;
  }



  public void setPublicIp(String publicIp) { 
    this.publicIp = publicIp;
  }

  public ServerIpmiLiveInfo allowedIp(String allowedIp) { 

    this.allowedIp = allowedIp;
    return this;
  }

  /**
   * Get allowedIp
   * @return allowedIp
   **/
  
  @Schema(description = "")
  
  public String getAllowedIp() {  
    return allowedIp;
  }



  public void setAllowedIp(String allowedIp) { 
    this.allowedIp = allowedIp;
  }

  public ServerIpmiLiveInfo clientUsername(String clientUsername) { 

    this.clientUsername = clientUsername;
    return this;
  }

  /**
   * Get clientUsername
   * @return clientUsername
   **/
  
  @Schema(description = "")
  
  public String getClientUsername() {  
    return clientUsername;
  }



  public void setClientUsername(String clientUsername) { 
    this.clientUsername = clientUsername;
  }

  public ServerIpmiLiveInfo clientPassword(String clientPassword) { 

    this.clientPassword = clientPassword;
    return this;
  }

  /**
   * Get clientPassword
   * @return clientPassword
   **/
  
  @Schema(description = "")
  
  public String getClientPassword() {  
    return clientPassword;
  }



  public void setClientPassword(String clientPassword) { 
    this.clientPassword = clientPassword;
  }

  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    ServerIpmiLiveInfo serverIpmiLiveInfo = (ServerIpmiLiveInfo) o;
    return Objects.equals(this.text, serverIpmiLiveInfo.text) &&
        Objects.equals(this.publicIp, serverIpmiLiveInfo.publicIp) &&
        Objects.equals(this.allowedIp, serverIpmiLiveInfo.allowedIp) &&
        Objects.equals(this.clientUsername, serverIpmiLiveInfo.clientUsername) &&
        Objects.equals(this.clientPassword, serverIpmiLiveInfo.clientPassword);
  }

  @Override
  public int hashCode() {
    return Objects.hash(text, publicIp, allowedIp, clientUsername, clientPassword);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class ServerIpmiLiveInfo {\n");
    
    sb.append("    text: ").append(toIndentedString(text)).append("\n");
    sb.append("    publicIp: ").append(toIndentedString(publicIp)).append("\n");
    sb.append("    allowedIp: ").append(toIndentedString(allowedIp)).append("\n");
    sb.append("    clientUsername: ").append(toIndentedString(clientUsername)).append("\n");
    sb.append("    clientPassword: ").append(toIndentedString(clientPassword)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
